### PR TITLE
[Hs-209] Update choose role view

### PR DESCRIPTION
**Pull request by DongNguyenQ**

### CODE REVIEW

**Review by hophongit98 — COMMENTED** [review]
Could you pls help to create a folder chooserole inside signup folder?
Move all the logic choose_role_view into chooserole. Btw rename choose_role_view -> choose_role_screen

File: lib/features/sign_up/view/widgets/choosing_roles_view.dart [hunk 1]
```diff
@@ -26,92 +26,87 @@ class ChoosingRoleViewBody extends StatelessWidget {
...
+        },
+        child: Scaffold(
+            appBar: AppBar(
+              elevation: 0,
+              backgroundColor: HSColor.background,
+              actions: [
+                Container(
+                    margin: const EdgeInsets.only(right: HsDimens.spacing8),
+                    child: IconButton(
+                      icon: const Icon(Icons.close, color: HSColor.onSurface),
```
Comment (suesitran) [inline lib/features/sign_up/view/widgets/choosing_roles_view.dart:51]: Please use icon from design, and set correct icon size

File: lib/features/sign_up/view/widgets/choosing_roles_view.dart [hunk 2]
```diff
@@ -26,92 +26,87 @@ class ChoosingRoleViewBody extends StatelessWidget {
...
+        if (state is SubmitRoleSucceedState) {
+          context.goToHome();
+        }
+        if (state is SubmitRoleFailedState) {
+          // TODO: Implement failure scenario
+        }
+      },
+      child: WillPopScope(
+        onWillPop: () async {
+          context.goToHome();
```
[the remark this answers is not in the document]
Comment (DongNguyenQ) [reply lib/features/sign_up/view/widgets/choosing_roles_view.dart:40]: @suesitran ViewPopScope to handle `Scenario 6: Close screen "Choose” by back hardware button (Android)`

**Comment by DongNguyenQ** [comment]
> Could you pls help to create a folder chooserole inside signup folder? Move all the logic choose_role_view into chooserole. Btw rename choose_role_view -> choose_role_screen

@suesitran Please help to confirm this. 
Currently signup folder is as below
sign_up
- View
- View_model
- choose_role (If we do Phong's suggestion. It will be like this ? )
   - view
     - widgets/
     - choose_role_screen 
   - view_model

**Review by suesitran — APPROVED** [review]

**Pull request merged**